core/src/test/java/org/itsallcode/openfasttrace/core/serviceloader/ServiceOriginTest.java

```diff
@@ -0,0 +1,74 @@
+package org.itsallcode.openfasttrace.core.serviceloader;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import java.net.URLClassLoader;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+// [utest->dsn~plugins.loading~1]
+@ExtendWith(MockitoExtension.class)
+class ServiceOriginTest
+{
+    @Test
+    void closesUrlClassLoader(@Mock final URLClassLoader classLoaderMock)
+            throws Exception
+    {
+        final ServiceOrigin serviceOrigin = new ServiceOrigin.JarFileOrigin(Collections.emptyList(), classLoaderMock);
+        serviceOrigin.close();
+        verify(((AutoCloseable) classLoaderMock)).close();
+        verifyNoMoreInteractions(classLoaderMock);
```

Comment:
:+1: Good. Normally I am not a fan of `verifyNoMoreInteractions`, but here it is absolutely justified.